### PR TITLE
bug/5854-Dylan-AppointmentCalendarGeolocationAddress

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
Modified iOS rnCalendar files to include lat and long and add location accordingly, updated android to use the address instead of the name if provided. Links are clickable in the Calendar to take users to maps to navigate to the location of the appointment.(compatible with iOS and Google Calendar)

## Screenshots/Video

https://github.com/department-of-veterans-affairs/va-mobile-app/assets/87150991/1af2fe19-2005-4e83-9e89-941d5c756f8e

https://github.com/department-of-veterans-affairs/va-mobile-app/assets/87150991/3a90c02f-90e3-4961-8e2a-02d3a9226859

## Testing
yarn test

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
What would be actually helpful to the veteran, would be supplying the actual location info (address or lat/long, whatever is supported by the 'add to calendar' APIs), so they could get to their appointment on the day of, without needing to re-open the VA mobile app to use the 'get directions' functionality to get there.

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: VAMobile/src/screens/HealthScreen/Appointments/UpcomingAppointments/UpcomingAppointmentDetails.tsx [hunk 1]
```diff
@@ -104,7 +105,9 @@ const UpcomingAppointmentDetails: FC<UpcomingAppointmentDetailsProps> = ({ route
       title: getTranslation(isCovidVaccine ? 'upcomingAppointments.covidVaccine' : AppointmentTypeToID[appointmentType], t),
       startTime: getEpochSecondsOfDate(startDateUtc),
       endTime: getEpochSecondsOfDate(endTime),
-      location: name || '',
+      location: isIOS() && lat && long ? name || '' : address ? address.street + ' ' + address.city + ', ' + address.state + ' ' + address.zipCode : name || '',
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/Appointments/UpcomingAppointments/UpcomingAppointmentDetails.tsx:108]:
```suggestion
      location: isIOS() && lat && long ? name || '' : address ? `${address.street} ${address.city}, ${address.state} ${address.zipCode}` : name || '',
```
Template literals make it easier to read

File: VAMobile/src/store/api/demo/mocks/appointments.json [hunk 2]
```diff
@@ -96,13 +96,13 @@
               "id": null,
               "name": "Outpatient Clinic",
               "address": {
-                "street": "2341 North Ave",
-                "city": "Commerce",
+                "street": "11201 Benton Street",
+                "city": "Loma Linda",
```
Comment (theodur) [inline VAMobile/src/store/api/demo/mocks/appointments.json:100]: This is my local VA facility 🙂

**Review by dumathane — DISMISSED** [review]

**Review by theodur — DISMISSED** [review]

**Comment by dumathane** [comment]
image
In examples where the address was full of nulls we were having issues via Rachael's comment on the ticket. This last commit should fix it but i'm open to syntax suggestions.

File: VAMobile/src/screens/HealthScreen/Appointments/UpcomingAppointments/UpcomingAppointmentDetails.tsx [hunk 3]
```diff
@@ -104,7 +105,14 @@ const UpcomingAppointmentDetails: FC<UpcomingAppointmentDetailsProps> = ({ route
       title: getTranslation(isCovidVaccine ? 'upcomingAppointments.covidVaccine' : AppointmentTypeToID[appointmentType], t),
       startTime: getEpochSecondsOfDate(startDateUtc),
       endTime: getEpochSecondsOfDate(endTime),
-      location: name || '',
+      location:
+        isIOS() && lat && long
```
Comment (alexandec) [inline VAMobile/src/screens/HealthScreen/Appointments/UpcomingAppointments/UpcomingAppointmentDetails.tsx:109]: I find nested ternaries hard to read, especially when combined with booleans expressions. Would recommend a refactor (curious if others agree)

**Review by alexandec — CHANGES_REQUESTED** [review]

File: VAMobile/src/screens/HealthScreen/Appointments/UpcomingAppointments/UpcomingAppointmentDetails.tsx (hunk 3 above)
Comment (theodur) [reply VAMobile/src/screens/HealthScreen/Appointments/UpcomingAppointments/UpcomingAppointmentDetails.tsx:109]: I agree, I think refactoring this into a helper function and separating the logic would help with readability and maintainability

File: VAMobile/src/store/api/demo/mocks/appointments.json (hunk 2 above)
Comment (alexandec) [reply VAMobile/src/store/api/demo/mocks/appointments.json:100]: Nice, I went to the one in Cheyenne, WY for my PIV card. That one shows up in demo mode as well

**Review by alexandec — APPROVED** [review]

**Review by theodur — APPROVED** [review]

**Pull request merged**